src/functions-and-arrays.js

```diff
@@ -1,20 +1,113 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers (num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num1 < num2) {
+    return num2;
+  } else {
+    return num1; 
+  }
+}
+
 // Iteration #2: Find longest word
+
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord (arr) {
+  if (!arr.length) { return null; }
+  let word = ""; 
+  arr.forEach((element) => {
+    if (word.length < element.length) {
+      word = element;
+    }
+  })
+  return word; 
+}
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers (arr) {
+  if (!arr.length) { return 0; }
+  let sum = 0; 
+  arr.forEach((element) => {
+    sum += element; 
+  })
+  return sum; 
+}
+
+// Bonus - Iteration #3.1
+
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+function sum (arr) {
+  if (!arr.length) { return 0; }
+  let sum = 0;
+  arr.forEach((element) => {
+    if (typeof element === "number") {
+      sum += element; 
+    } else if (typeof element === "string") {
+      sum += element.length; 
+    } else if (element === true) {
+      sum += 1; 
+    } else if (typeof element === "object" || typeof element === "array") {
```

Comment:
In this line, you should only check for the first statement, since an array is just a "special" object! For more information you can check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures